web/controllers/pair_retro_controller.ex

```diff
@@ -13,28 +13,48 @@ defmodule Pairmotron.PairRetroController do
   end
 
   def new(conn, %{"pair_id" => pair_id}) do
-    conn = assign(conn, :projects, Repo.all(Project))
-    current_user = conn.assigns[:current_user]
-    changeset = PairRetro.changeset(%PairRetro{}, %{pair_id: pair_id, user_id: current_user.id}, nil)
-    render(conn, "new.html", changeset: changeset)
+    current_user = conn.assigns.current_user
+
+    pair = Pair.pair_with_users(pair_id) |> Repo.one
+
+    cond do
+      is_nil(pair) ->
+        redirect_and_flash_error(conn, "You cannot create a retrospective for non-existent pair")
+      not current_user.id in Enum.map(pair.users, &(&1.id)) ->
+        redirect_and_flash_error(conn, "You cannot create a retrospective for a pair you are not in")
+      true ->
+        conn = assign(conn, :projects, Repo.all(Project))
+        current_user = conn.assigns[:current_user]
+        changeset = PairRetro.changeset(%PairRetro{}, %{pair_id: pair_id, user_id: current_user.id}, nil)
+        render(conn, "new.html", changeset: changeset)
+    end
   end
 
   def create(conn, %{"pair_retro" => pair_retro_params}) do
-    user_id = parameter_as_integer(pair_retro_params, "user_id")
-    current_user = conn.assigns[:current_user]
-    if current_user.id == user_id || user_id == 0 || is_admin?(current_user) do
-      earliest_pair_date = earliest_pair_date_from_params(pair_retro_params)
-      changeset = PairRetro.changeset(%PairRetro{}, pair_retro_params, earliest_pair_date)
-      case Repo.insert(changeset) do
-        {:ok, _pair_retro} ->
-          conn
-          |> put_flash(:info, "Pair retro created successfully.")
-          |> redirect(to: pair_retro_path(conn, :index))
-        {:error, changeset} ->
-          render(conn, "new.html", changeset: changeset)
-      end
-    else
-      redirect_not_authorized(conn, pair_retro_path(conn, :index))
+    current_user = conn.assigns.current_user
+
+    pair_id = parameter_as_integer(pair_retro_params, "pair_id")
+    pair = Pair.pair_with_users(pair_id) |> Repo.one
+
+    cond do
+      is_nil(pair) ->
+        redirect_and_flash_error(conn, "You cannot create a retrospective for non-existent pair")
+      not current_user.id in Enum.map(pair.users, &(&1.id)) ->
+        redirect_and_flash_error(conn, "You cannot create a retrospective for a pair you are not in")
+      true ->
+        implicit_params = %{"user_id" => conn.assigns.current_user.id}
+        final_params = pair_retro_params |> Map.merge(implicit_params)
```

Comment:
Decided to make the user implicitly the logged in user. It's safer, results in cleaner code, and the argument against doing so was largely based on giving admins the ability to interact with retros before ex_admin was added.